libqtile/backend/x11/xcore.py

```diff
@@ -328,7 +323,35 @@ def get_valid_timestamp(self):
         """Get a valid timestamp, i.e. not CurrentTime, for X server.
 
         It may be used in cases where CurrentTime is unacceptable for X server."""
-        return self._last_event_timestamp
+        # do a zero length append to get the time offset as suggested by ICCCM
+        # https://tronche.com/gui/x/icccm/sec-2.html#s-2.1
+        # we do this on a separate connection since we can't receive events
+        # without returning control to the event loop, which we can't do
+        # because the event loop (via some window event) wants to know the
+        # current time.
+        conn = None
+        try:
+            conn = xcbq.Connection(self._display_name)
+            conn.default_screen.root.set_attribute(
+                eventmask=xcffib.xproto.EventMask.PropertyChange
+            )
+            conn.conn.core.ChangePropertyChecked(
+                xcffib.xproto.PropMode.Append,
+                self._root.wid,
+                self.conn.atoms["WM_CLASS"],
+                self.conn.atoms["STRING"],
+                8,
+                0,
+                "",
+            )
+            while True:
+                event = conn.conn.wait_for_event()
```

Comment:
added some debug messages and Qtile hangs in this line after starting Jetbrains app

Reply:
Ok, thanks. Seems like something about the masking is wrong :(